pkg/storage/mvcc.go

```diff
@@ -86,6 +91,17 @@ var MaxConflictsPerLockConflictError = settings.RegisterIntSetting(
 	settings.WithName("storage.mvcc.max_conflicts_per_lock_conflict_error"),
 )
 
+// MaxBytesPerLockConflictError sets maximum number of locks under this
+// byte threshold returned in LockConflictError in operations that return multiple
+// locks per error.
```

Comment:
We should probably name this in terms of "target" rather than "max", since we will overshoot it (i.e. we will keep collecting intents until we are >= the target, and then return everything we have including the one which exceeded the limit, so this will almost always be > target). This is in line with other options like `ScanOptions.TargetBytes` and `ExportOptions.TargetBytes` which also overshoot.

This does put some responsibility on callers to make sure they don't then generate intent resolution commands that exceed the max Raft command size (consider e.g. a single intent that is 64 MB large at the end of a batch that already has 8 MB of intents). The intent resolver also uses a target batch size, rather than a strict max, but that can be addressed separately -- this is an improvement in any case.